Tests/Set-BMVariable.Tests.ps1

```diff
@@ -285,6 +299,16 @@ Describe 'Set-BMVariable' {
         ThenNoErrorWritten
     }
 
+    It 'should convert map to OtterScript map' {
+        WhenSettingVariable 'GlobalVar' -WithValue @{ 'hello' = 'world' }
+        ThenVariableSet 'GlobalVar' -To '%(hello: world)' -Raw
+    }
+
+    It 'should convert array to OtterScript vector' {

```

Comment:
Add a test for when an object that isn't a string, array, or hashtable is passed to `-WithValue`.